backend/src/providers/transactions.ts

```diff
@@ -126,6 +130,127 @@ export const getTransactionInfo = async (
   return transaction[0] || null;
 };
 
+export const getTransactionDetails = async (
+  transactionHash: string
+): Promise<TransactionDetails | null> => {
+  const transactionBaseInfo = await getTransactionInfo(transactionHash);
+
+  if (!transactionBaseInfo) {
+    throw new Error(`No hash ${transactionHash} found`);
+  }
+  const transactionInfo = await sendJsonRpc("EXPERIMENTAL_tx_status", [
+    transactionBaseInfo.hash,
+    transactionBaseInfo.signerId,
+  ]);
+  const blockHashes = transactionInfo.receipts_outcome.map(
+    (receipt) => receipt.block_hash
+  );
+  const blockHeights = await getBlockHeightsByHashes(blockHashes);
+  const includedInBlockMap = blockHeights.reduce((acc, block) => {
+    acc.set(block.blockHash, {
+      blockHeight: block.blockHeight,
+      blockTimestamp: block.blockTimestamp,
+    });
+    return acc;
+  }, new Map());
+  const receiptsOutcome = transactionInfo.receipts_outcome.map((receipt) => ({
+    id: receipt.id,
+    proof: receipt.proof,
+    outcome: receipt.outcome,
+    includedInBlock: {
+      hash: receipt.block_hash,
+      height: includedInBlockMap.get(receipt.block_hash).blockHeight,
+      timestamp: includedInBlockMap.get(receipt.block_hash).blockTimestamp,
+    },
+  }));
+
+  if (
```

Comment:
Lot of this code is copy-pasted from current transaction fetch implementation.
Can we refactor our the duplicated part?

Reply:
I think it's better to do in #1048